### PR TITLE
Fix compilation and execution on i386 (and possibly other architectures)

### PR DESCRIPTION
**Pull request by lzaoral**
The `va_arg` fix in 5f21113 is not very elegant but variadic templates + perfect forwarding would probably be an overkill for this.

### CODE REVIEW

**Comment by lzaoral** [comment]
I see that the rewrite of variadic functions to variadic templates were proposed in #213. I'll look into that.

**Comment by lzaoral** [comment]
Ok, the PR should be ready now. However, the the Travis status is still missing.

File: include/dg/PointerAnalysis/PSNode.h [hunk 1]
```diff
@@ -744,6 +706,76 @@ class PSNodeJoin : public PSNode {
...
+    using type = PSNodeRet;
+};
+
+template<>
+struct GetNodeType<PSNodeType::CALL_RETURN>
+{
+    using type = PSNodeCallRet;
+};
+
+
```
Comment (lzaoral) [inline include/dg/PointerAnalysis/PSNode.h:778]: C++17 note: This whole part can be replaced with `if constexpr` in `dg::pta::PointerGraph::nodeFactory`

**Comment by lzaoral** [comment]
Oh, snap. It seems that I have used some C++14 features.

**Comment by mchalupa** [comment]
I know it is annoying, but let's stick to C++11 for now. I plan to switch to C++14 after version 1.0 is released, which is hopefully not that far in the future.

**Comment by lzaoral** [comment]
Alright, the code should be C++11 only now (though uglier :/) with comments for possible improvement in the future.

**Comment by mchalupa** [comment]
Awesome, thanks!

**Pull request merged**